WeakAuras/Transmission.lua

```diff
@@ -316,39 +270,152 @@ ChatFrame_AddMessageEventFilter("CHAT_MSG_BN_WHISPER_INFORM", filterFunc)
+buttonAnchor:RegisterEvent("PLAYER_REGEN_DISABLED")
+
+local thumbnailAnchor = CreateFrame("FRAME",  "WeakAurasTooltipThumbnailFrame", buttonAnchor)
+thumbnailAnchor:SetSize(40,40)
+thumbnailAnchor:SetPoint("TOPRIGHT", -27, -8)
+
+local importButton = CreateFrame("BUTTON", "WeakAurasTooltipImportButton", buttonAnchor, "UIPanelButtonTemplate")
+importButton:SetPoint("BOTTOMRIGHT", -8, 8)
+
+
+local showCodeButton = CreateFrame("BUTTON", "WeakAurasTooltipImportButton", buttonAnchor, "UIPanelButtonTemplate")
+showCodeButton:SetPoint("BOTTOMLEFT", 8, 8)
+showCodeButton:SetText(L["Show Code"])
+showCodeButton:SetWidth(90)
+
+local checkButtons, keyToButton, pendingData = {}, {}, {}
+for _, key in pairs(WeakAuras.internal_fields) do
+  keyToButton[key] = false
+end
+
+for index,category in pairs(WeakAuras.update_categories) do
+  local button = CreateFrame("checkButton", "WeakAurasTooltipCheckButton"..index, buttonAnchor, "ChatConfigCheckButtonTemplate")
+  for k, v in pairs(category) do
+    button[k] = v
   end
-  if(ItemRefTooltip.WeakAuras_Tooltip_Button2) then
-    ItemRefTooltip.WeakAuras_Tooltip_Button2:Hide();
+  button.func = function()
+    WeakAurasSaved.update_preferences[button.name] = button:GetChecked()
+    if button:GetChecked() ~= button.checkedStateForCopy then
+      pendingData.buttonsChecked = min(pendingData.buttonsChecked + 1, #checkButtons)
+    else
+      pendingData.buttonsChecked = max(pendingData.buttonsChecked - 1, 0)
+    end
+    WeakAuras.RefreshTooltipButtons()
   end
-  if(ItemRefTooltip.WeakAuras_Desc_Box) then
-    ItemRefTooltip.WeakAuras_Desc_Box:Hide();
+  checkButtons[index] = button
+  checkButtons[button.name] = button
+  for _, key in pairs(category.fields) do
+    keyToButton[key] = button
   end
-  ShowUIPanel(ItemRefTooltip);
-  if not ItemRefTooltip:IsVisible() then
-    ItemRefTooltip:SetOwner(UIParent, "ANCHOR_PRESERVE");
+end
+
+-- Ideally, we would not add an __index method to this
+-- But that would greatly bloat the update_category for display
+-- A better solution would be to refactor the data
+-- Such that all of the display tab settings are in their own subtable
+setmetatable(keyToButton,{__index = function(_, key) return key and checkButtons["display"] end,})
+
+local function install(data, oldData, patch, forceCopy)
+  if not oldData or forceCopy then
+    if oldData ~= "added" or not checkButtons["newchildren"]:GetChecked() then
+      if forceCopy then
+        data.uid = nil
+      end
+      data.id = WeakAuras.FindUnusedId(data.id)
+      WeakAuras.Add(data)
+    end
+  elseif not data then -- unmatched old child which is being kept
+    data = oldData
+  else
+    patch = patch or {}
+      for key in pairs(patch) do
+        local checkButton = keyToButton[key]
+        if checkButton and not checkButton:GetChecked() then
+          patch[key] = nil
+        end
+      end
+    if patch.id then
+        patch.id = WeakAuras.FindUnusedId(patch.id)
```

Comment:
Indentation error. Also there's a corner case here, which I think we should do nothing about and just ignore, but which I'll try to explain anyway:

Imagine a group:
G A B with uids 0 1 2.

Importing a group:
G B A with uids 0 1 2

That is in the new version of the group the author renamed A to B, and B to A. For both A and B, patch.id will be set.

On importing B, the code will run FindUnusedId("B"), which finds the not yet renamed old B, and thus chooses "B 2" as the new name.

Again, I think there are limits to what should be supported, and this falls outside of it.